### PR TITLE
feat: refine mobile turn action dock

### LINKED ISSUE #129: feat: redesign shared turn action bar presets and restore bottom check contract
## Reviewer attention
- This work changes the shared turn-action dock across desktop and mobile, but the final compact-mobile contract is narrower than the original preset-row expansion. Review the mobile flow as a dedicated two-column dock with a dock-level raise popover rather than a flat row of many standalone raise buttons.
- The bottom direct-action contract is intentionally stable again: `Check` remains beside `Fold` in all states and expresses legality through disabled state instead of swapping to `Call`.
- The branch also carries two follow-up correctness fixes tied to the same mobile/action workflow: authoritative `minRaise` recovery on room restore/reconnect, and hand-scoped `Your Cards` auto-hide after results until the next hand begins.

## Current evidence / code anchors
- `poker-client/src/components/poker/turn-action-dock.tsx` now keeps `Check` and `Fold` fixed in the footer, renders `Call` / `Min Bet` as standalone amount-composer actions when legal, and opens extra raise actions through a compact mobile popover rendered above the whole action dock.
- `poker-client/src/components/poker/use-anchored-popover.ts` now supports `fixed` positioning so the compact mobile raise popover can escape lower stacking contexts.
- `poker-client/src/index.css` and `poker-client/src/components/poker/your-cards-flyout.tsx` reserve a non-desktop bottom-safe lane for the cards and keep the cards from covering the active bottom UI.
- `poker-server/src/events/events.gateway.ts` now injects authoritative `minRaise` into sanitized active-hand snapshots so reconnect/refresh restores do not fall back to an illegal minimum raise.
- `poker-client/src/components/GameRoom.tsx` keeps `Your Cards` hidden once hand results begin and restores the flyout at the next hand boundary.

## Problem / request
The turn action dock had drifted away from the intended table UX. The bottom-left direct action swapped from `Check` to `Call`, recent changes made the `Check` contract easier to break, common raise shortcuts were missing from the amount composer, compact mobile layouts became crowded, reconnect/refresh could surface an illegal minimum raise, and the non-desktop `Your Cards` overlay could collide with live bottom actions.

## Why this matters
The turn action dock is the highest-frequency interaction surface in the room. Players need stable footer actions for muscle memory, fast legal raise shortcuts, predictable mobile behavior, and trustworthy betting constraints after reconnects. The hero cards also need a systematic mobile position so they never hide action controls or the table state.

## Desired outcome
Ship a shared turn dock where `Check` stays beside `Fold`, `Check` is disabled when illegal, desktop keeps the wider action-composer treatment, and compact mobile uses a two-column dock that leaves only the essential standalone actions visible while opening the remaining legal raise actions in a dock-level popover above the action dock. Keep illegal non-all-in raise actions hidden, preserve existing tray review/confirm and drag-to-pot behavior, recover authoritative `minRaise` on room restore/reconnect, and auto-hide `Your Cards` when a hand finishes until the next hand starts.

## Contracted scope
- Restore the intended bottom direct-action behavior in `TurnActionDock` so `Check` no longer swaps to `Call`.
- Keep `Check` visible beside `Fold` in all turn states and disable it whenever checking is not legal.
- Keep `Call` in the amount-composer flow instead of the footer.
- Add `1/3 Pot`, `1/2 Pot`, `Pot`, `Min Raise`, and `All-in` shortcuts to the shared betting composer where legal.
- Keep `Min Raise` immediately after `Call` in the non-compact composer ordering.
- Hide illegal non-all-in raise presets instead of rendering them disabled.
- Use the wider desktop dock treatment in the desktop preview/layout.
- On compact mobile, keep the two-column tray/actions layout, leave only the legal standalone `Call` or `Min Bet` action outside the raise menu, and show a `Raise` trigger only when at least one additional legal raise action exists.
- Render the compact mobile raise menu in a popover above the whole action dock, with explicit dismiss affordance, so it stays clear of `Your Cards`.
- Keep custom amount input and `Clear` in the dock below the compact mobile action row.
- Reserve non-desktop bottom-safe table space so `Your Cards` stays at the bottom edge without overlapping the felt, turn dock, or operation action bar.
- Auto-hide `Your Cards` when a hand finishes and the hand-results flow starts, and keep it hidden until the next new hand.
- Restore authoritative `minRaise` in sanitized room snapshots so reconnect/refresh preserves legal minimum raise behavior.
- Preserve current betting semantics, drag-to-pot support, quick-confirm behavior, and backend action contracts.

## Constraints / non-goals
- Do not change server APIs, room event contracts, or betting rules.
- Do not remove the existing click-first desktop tray review/confirm flow.
- Do not redesign unrelated chat-rail or showdown operation-bar behavior beyond spacing needed to avoid overlap.
- Do not reintroduce a compact mobile layout that depends on many standalone preset buttons.

## Acceptance criteria
- [x] The bottom direct-action footer always renders `Check` beside `Fold`; it no longer swaps that slot to `Call` in any turn state.
- [x] The bottom `Check` button is disabled whenever checking is illegal and cannot dispatch a check action in that state.
- [x] `Call` remains available through the amount-composer flow rather than as a footer replacement.
- [x] The shared betting composer includes `Call`, `Min Raise`, `1/3 Pot`, `1/2 Pot`, `Pot`, and `All-in` where those actions resolve legally.
- [x] Illegal non-all-in raise presets are hidden instead of rendered disabled.
- [x] The desktop preview/layout uses the wider dock treatment instead of the earlier narrow Storybook frame.
- [x] Compact mobile keeps the two-column tray/actions layout and only shows a `Raise` trigger when at least one extra legal raise action exists.
- [x] The compact mobile raise popover renders above the whole action dock, clears `Your Cards`, and exposes explicit dismiss behavior.
- [x] The custom amount input and `Clear` remain available below the compact mobile action row.
- [x] On non-desktop layouts, `Your Cards` stays in a bottom-safe lane and does not overlap the felt/table, turn dock, or operation action bar.
- [x] `Your Cards` auto-hides when a hand is finished and remains hidden until the next hand starts.
- [x] Room restore/reconnect preserves the authoritative legal `minRaise` for the active hand.
- [x] Targeted automated coverage demonstrates the restored footer contract, legal preset filtering, reconnect-safe minimum raise recovery, and hand-finished card visibility behavior.

## External setup dependencies
- None.

## Affected repo area
- `poker-client/src/components/GameRoom.tsx`
- `poker-client/src/components/poker/mobile-overlay-layout.ts`
- `poker-client/src/components/poker/table-shell.tsx`
- `poker-client/src/components/poker/turn-action-dock.tsx`
- `poker-client/src/components/poker/turn-action-presets.ts`
- `poker-client/src/components/poker/your-cards-flyout.tsx`
- `poker-client/src/components/poker/use-anchored-popover.ts`
- `poker-client/src/i18n/messages.ts`
- `poker-client/src/index.css`
- `poker-client/src/components/poker/*.spec.ts`
- `poker-server/src/events/events.gateway.ts`
- `poker-server/test/unit/events.gateway.showdown-muck.spec.ts`
- `poker-server/test/e2e/comprehensive-poker.spec.ts`

## Owning layer or module
Shared turn-action dock and non-desktop overlay behavior in `poker-client`, plus authoritative active-hand sanitization in `poker-server`.

## Validation commands
- `pnpm --dir poker-client test -- --run src/components/poker/turn-action-dock.spec.ts`
- `pnpm --dir poker-server test -- --run test/unit/events.gateway.showdown-muck.spec.ts`
- `PW_FRONTEND_PORT=5188 PW_BACKEND_PORT=3015 pnpm --dir poker-server run test:e2e:playwright --project comprehensive-e2e --workers=1 --grep "8\.2c:|8\.2d:|8\.12h|8\.9b:" --reporter=line`

## Links / references
- Regression introduced by commit `a411d9d`
- Related desktop action-dock follow-up: #124
- Built on the desktop action-dock contract from #115 / PR #118
- Active PR: #132

### PR DESCRIPTION
**Pull request by grepug**
## Summary

- Refine the compact mobile action dock around a dock-level raise popover while preserving the desktop check/fold footer contract
- Keep reconnect-safe min-raise recovery and hand-scoped Your Cards hiding aligned with the final shipped mobile UX

## Linked Issue

Closes #129

## Closeout Checklist

- [x] Re-read the linked issue after implementation
- [x] Reconciled the canonical plan comment checklist against the shipped code
- [x] No unchecked implementation checklist items remain in the issue body or canonical plan comment
- [x] This PR inherits the linked issue milestone when one exists

## Checklist Audit

- Issue body unchecked items: 0
- Canonical plan comment blocking unchecked items: 0
- Canonical plan comment non-blocking unchecked items: 0
- Canonical plan comment: https://github.com/grepug/Poker/issues/129#issuecomment-4219997625

## Validation

- pnpm --dir poker-client test -- --run src/components/poker/turn-action-dock.spec.ts
- pnpm --dir poker-server test -- --run test/unit/events.gateway.showdown-muck.spec.ts
- PW_FRONTEND_PORT=5188 PW_BACKEND_PORT=3015 pnpm --filter poker-server test:e2e:playwright:smoke

### CODE REVIEW

File: poker-client/src/components/poker/turn-action-dock.tsx [hunk 1]
```diff
@@ -54,13 +50,31 @@ type TurnActionDockProps = {
...
+  if (callAmount > 0) {
+    return {
+      standalonePresets: trayPresetButtons.filter((preset) => preset.key === "call"),
+      menuPresets: trayPresetButtons.filter((preset) => preset.key !== "call"),
+    };
+  }
+
+  return {
+    standalonePresets: trayPresetButtons.filter((preset) => preset.key === "min-raise"),
+    menuPresets: trayPresetButtons.filter((preset) => preset.key !== "min-raise"),
```
[the remark this answers is not in the document]
Comment (grepug) [reply poker-client/src/components/poker/turn-action-dock.tsx:75]: Addressed. I kept the product rule that only `Call` / `Min Bet` stay standalone on compact mobile, but fixed the misleading edge case by renaming the trigger when the menu only contains `All-In`. Added a unit test for the all-in-only menu state and pushed the update in a80e2c2.

File: poker-client/src/index.css [hunk 2]
```diff
@@ -1702,22 +1708,60 @@ body {
...
+  gap: 0.32rem;
+  flex: 1 1 16rem;
+  min-width: 0;
+}
+
+.chip-composer-dock__mobile-presets {
+  display: grid;
+  grid-template-columns: repeat(2, minmax(0, 1fr));
   gap: 0.32rem;
 }
```
[the remark this answers is not in the document]
Comment (grepug) [reply poker-client/src/index.css:1742]: Not taking this change. This comment is stale relative to the final accepted UX: the compact mobile dock was intentionally changed to a two-column layout with only `Call` / `Min Bet` outside the popover, while raise-related actions stay collapsed behind the inline trigger. That matches the current product decision and follow-up user direction in this branch.

**Comment by grepug** [comment]
Follow-up from review: tightened the compact-mobile edge case further in `cd69695`.

When there is no standalone compact preset and only one remaining preset exists, the dock now surfaces that action directly instead of routing the user through a one-item popover. This keeps the normal two-column raise-popover design, but removes the short-stack/all-in dead-end noted in review.

Validation:
- `pnpm --dir poker-client test -- --run src/components/poker/turn-action-dock.spec.ts`

File: poker-client/src/components/GameRoom.tsx [hunk 3]
```diff
@@ -3027,6 +2983,14 @@ const useGameRoomElement = () => {
     setShowHandResultsModal(true);
   }, [lastHandResult]);
 
+  useEffect(() => {
+    if (!lastHandResult) {
+      return;
+    }
+
+    setIsCardsFlyoutOpen(false);
+  }, [lastHandResult, setIsCardsFlyoutOpen]);
```
[the remark this answers is not in the document]
Comment (grepug) [reply poker-client/src/components/GameRoom.tsx:2992]: Valid comment. Fixed by gating `shouldRenderCardsFlyout` behind `!lastHandResult`, so the flyout and its toggle no longer render during the hand-results phase and stay hidden until the next hand clears `lastHandResult`.\n\nValidation:\n- `pnpm --dir poker-client test -- --run src/components/poker/turn-action-dock.spec.ts`\n- `PW_FRONTEND_PORT=5188 PW_BACKEND_PORT=3015 pnpm --dir poker-server run test:e2e:playwright --project comprehensive-e2e --workers=1 --grep "8\\.9b:" --reporter=line`
